From 9d8f5bbd2478c448b551d758ce84026241b4eaac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=A9mie=20Faucher-Goulet?=
 <jeremiefauchergoulet@hotmail.com>
Date: Fri, 6 Aug 2021 14:31:02 -0400
Subject: [PATCH] Fix issue #4491: USB device reset for Privileged VMs (with
 PCI device attached) is not working due to bad argument '-r'
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jérémie Faucher-Goulet <jeremiefauchergoulet@hotmail.com>
---
 scripts/usb_reset.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/usb_reset.py b/scripts/usb_reset.py
index 5a57602277..eb87a4873a 100755
--- a/scripts/usb_reset.py
+++ b/scripts/usb_reset.py
@@ -66,7 +66,7 @@ def parse_arg():
                         help="specify the domid of the VM")
     attach.add_argument("-p", dest="pid", type=int, required=True,
                         help="the process id of QEMU")
-    attach.add_argument("-r", dest="reset_only", type=int,
+    attach.add_argument("-r", dest="reset_only", action="store_true",
                         help="reset device only, for privileged mode")
 
     detach = subparsers.add_parser("detach", help="detach a USB device")
